### PR TITLE
feat(file-watching): add documentation and refresh of web components

### PR DESCRIPTION
**Pull request by pkkummermo**
This PR includes documentation for the custom components as well as allowing us to watch for changes when the custom component changes in terms of edit and creation. Using a TS Program
and a watchCompilerHost (while bypassing the compile), we can get an incremental change handler for ts files in the project using the tsconfig.json (if available).

This PR adresses https://github.com/aurelia/vscode-extension/issues/118, as well as the problem of needing to reload VSC for new/edited components.

### CODE REVIEW

File: src/server/main.ts [hunk 1]
```diff
@@ -195,11 +195,60 @@ async function handlefeatureToggles({ featureToggles, extensionSettings }: Aurel
...
+
+        console.log('>>> The extension found this many components:');
+        console.log(aureliaApplication.components.length);
+      };
+
+      const configPath = ts.findConfigFile(
+        /* searchPath */ "./",
+        ts.sys.fileExists,
+        "tsconfig.json"
+      );
```
Comment (hiaux0) [inline src/server/main.ts:211]:
```suggestion
      const configPath = ts.findConfigFile(
        /* searchPath */ "./",
        ts.sys.fileExists,
        "tsconfig.json"
      );
      
      if (configPath === undefined) {
        configPath = '../../tsconfig.json' // use config file from the extension as default
      }
```

Comment (pkkummermo) [reply src/server/main.ts:211]: Good idea. Perhaps we should have a dedicated config in the repo which is easier to track and change.

File: src/server/main.ts [hunk 2]
```diff
@@ -195,11 +195,64 @@ async function handlefeatureToggles({ featureToggles, extensionSettings }: Aurel
...
+      const updateAureliaComponents = async () => {
+        const fileProcessor = new ProcessFiles();
+        await fileProcessor.processPath(extensionSettings);
+        aureliaApplication.components = fileProcessor.components;
+
+        console.log('>>> The extension found this many components:');
+        console.log(aureliaApplication.components.length);
+      };
+
+      const configPath = ts.findConfigFile(
```
Comment (hiaux0) [inline src/server/main.ts:207]:
```suggestion
      let configPath = ts.findConfigFile(
```

Comment (hiaux0) [reply src/server/main.ts:207]: forgot that part, ups

**Pull request merged**